### PR TITLE
Created Rabin_Karp.cpp

### PR DESCRIPTION
**Pull request by Yash-Thakare22**
Implemented Rabin Karp Algorithm in C++

### CODE REVIEW

**Comment by HarshCasper** [comment]
Hello @YashThakare 

Thanks for making a Pull Request. I have added some reviews. Kindly resolve them before it is fit to be merged.

File: C++/search/Rabin_Karp.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#include<bits/stdc++.h>
+using namespace std;
+typedef long long int ll;
+//MODULAR EXPONENTIATION
+ll power(ll x,ll y,ll p)
+{
+	ll res=1;
+	x=x%p;
```
Comment (HarshCasper) [inline C++/search/Rabin_Karp.cpp:8]: Make use of Proper Variable Names, either Camel-Case/Snake-Case Naming Convention for better understanding of the Code.

File: C++/search/Rabin_Karp.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	while(y>0)
+	{
+		if(y&1)
```
Comment (HarshCasper) [inline C++/search/Rabin_Karp.cpp:11]: Add proper Comments for the User to understand the Code.

File: C++/search/Rabin_Karp.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		res=(res*x)%p;
+		y=y>>1;
+		x=(x*x)%p;
+	}
+	return res;
+}
+void search (string text,string pattern)
+{
+	ll m=text.size();
```
Comment (HarshCasper) [inline C++/search/Rabin_Karp.cpp:20]: Make use of proper Variable Names and format the expressions

File: C++/search/Rabin_Karp.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			ll j;
+			for( j=0;j<n;j++)
+			{
+				if(text[i+j]!=pattern[j])
+				break;
+			}
+			if(j==n)
+			cout<<"PATTERN FOUND AT "<<i<<endl;
+		}
+		if(i<m-n)
```
Comment (HarshCasper) [inline C++/search/Rabin_Karp.cpp:44]: Add Comments for a better understanding of the Code.

File: C++/search/Rabin_Karp.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	}
+}
+int main()
+{
+	string text,pattern;
+	getline(cin,text);
+	getline(cin,pattern);
+	search(text,pattern);
+	return 0;
+}
```
Comment (HarshCasper) [inline C++/search/Rabin_Karp.cpp:59]: Add a sample Input/Output for better understanding.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Kindly resolve them

**Comment by iamrajiv** [comment]
@YashThakare I think this is duplicate of #345

**Pull request closed without merging**